### PR TITLE
Several bugfixes in Docker build

### PR DESCRIPTION
**Pull request by ogarcia**
- Removed unnecessary variables
- Fixed symbolic links creation
- Fixed enter point
- Less intermediate containers

### CODE REVIEW

**Comment by Unknwon** [comment]
Thanks @ogarcia !

Have you tested those changes?

**Comment by ogarcia** [comment]
Yes!

If you want you can test it, i have an autobuild in https://hub.docker.com/r/ogarcia/gogs/
`docker pull ogarcia/gogs` do the magic :wink:

File: Dockerfile [hunk 1]
```diff
@@ -9,21 +9,13 @@ RUN echo "@edge http://dl-4.alpinelinux.org/alpine/edge/main" | tee -a /etc/apk/
...
-
-# Create git user for Gogs
-RUN adduser -D -g 'Gogs Git User' git -h /data/git/ -s /bin/sh && passwd -u git
-RUN echo "export GOGS_CUSTOM=/data/gogs" >> /etc/profile
-
 # Configure Docker Container
 VOLUME ["/data"]
 EXPOSE 22 3000
-CMD ["./docker/start.sh"]
+CMD ["docker/start.sh"]
```
Comment (Unknwon) [inline Dockerfile]: Only thing unsure is this, can you confirm `"docker/start.sh"` does exactly the same thing as `./docker/start.sh` does?

Comment (ogarcia) [inline Dockerfile]: Yes, cause you are in the workdir `/app/gogs/` and run `./docker/start.sh` or `docker/start.sh` is the same.

Comment (Unknwon) [inline Dockerfile]: Thanks your confirmation!

**Comment by Unknwon** [comment]
Thanks, merging...

File: Dockerfile (hunk 1 above)
Comment (ogarcia) [inline Dockerfile]: :+1:

File: docker/s6/gogs/run [hunk 2]
```diff
@@ -12,17 +11,16 @@ if ! test -d ~git/.ssh; then
...
-ln -sf /data/gogs/conf /app/gogs/conf
+cd /app/gogs
+
+ln -sf /data/gogs/log  ./log
+ln -sf /data/gogs/data ./data
 
 chown -R git:git /data /app/gogs ~git/
 
 export USER
-export USERNAME
```
Comment (0rax) [inline docker/s6/gogs/run]: Not sure if this line needs to be removed as this ENV var is checked during installation. I had a problem without having it. Will try to test the install process.

Thanks for the modification !

Comment (Unknwon) [inline docker/s6/gogs/run]: @0rax FYI, https://github.com/gogits/gogs/blob/master/modules/user/user.go#L11 either one of `USER` or `USERNAME` would work.

Comment (0rax) [inline docker/s6/gogs/run]: Thanks, was weird that it did not happened when USERNAME was not set :/ Will test it

Comment (ogarcia) [inline docker/s6/gogs/run]: @0rax I tested it in new install and running install and works perfect only with `USER`. As @Unknwon say, the `func CurrentUsername()` do the magic.

**Pull request merged**